### ISSUE TITLE
Migrate away from using the generate-groups script from k8s.io

### ISSUE THREAD

**Issue by roberthbailey** [googleforgames/agones#3659]
**Is your feature request related to a problem? Please describe.**

The `generate-groups` script was deprecated in Kubernetes 1.28 and will no longer exist with Kuberetes 1.30 ([source](https://groups.google.com/a/kubernetes.io/g/dev/c/Ix-ACY9DhEs)). We need to update our tooling to no longer depend upon it. 

**Describe the solution you'd like**
The replacement script is `kube_codegen.sh`. We should switch over to the new script.

**Describe alternatives you've considered**
We have to keep up with Kubernetes so there really isn't an alternative. 

**Additional context**
https://github.com/googleforgames/agones/blob/main/build/build-image/gen-crd-client.sh#L19

**Comment by markmandel** [comment]
Easiest way to check would be to run `make shell` and then look in `/go/src/k8s.io/code-generator`

**Comment by Kalaiselvi84** [comment]
Regarding the `generate-groups.sh` script in the `k8s.io/code-generator` directory, I noticed that the `kube_codegen.sh` script is not present there. Will the `kube_codegen.sh` script be added to this directory automatically, or is there a different process for obtaining and using it?

```
:/go/src/agones.dev/agones# cd /go/src/k8s.io/code-generator/
:/go/src/k8s.io/code-generator# ls
CONTRIBUTING.md  OWNERS     SECURITY_CONTACTS  code-of-conduct.md  examples            generate-internal-groups.sh  go.sum  pkg          tools.go
LICENSE          README.md  cmd                doc.go              generate-groups.sh  go.mod                       hack    third_party
```

**Comment by Kalaiselvi84** [comment]
The script `kube_codegen.sh` can be found in the `code-generator` repository at this link: https://github.com/kubernetes/code-generator/blob/master/kube_codegen.sh 

Could you please let me know the necessary steps we need to take?

**Comment by roberthbailey** [comment]
We can either vendor the code-generator repo (at the same branch version as we vendor client-go), or we can clone that github repository inside of our build image and execute it there.

**Comment by Kalaiselvi84** [comment]
Thanks @roberthbailey👍🏻

Just for my notes: We have cloned the code-generator repo in [here](https://github.com/googleforgames/agones/blob/main/build/build-image/Dockerfile#L61-L63). When Agones updates to Kubernetes 1.30, we will switch from `generate-groups.sh` to `kube-codegen.sh`.

**Comment by markmandel** [comment]
> When Agones updates to Kubernetes 1.30, we will switch from `generate-groups.sh` to `kube-codegen.sh`.

Just out of curiosity - I'm assuming the script doesn't exist in 1.29 ?

**Comment by Kalaiselvi84** [comment]
> Just out of curiosity - I'm assuming the script doesn't exist in 1.29 ?

I assumed that `generate-groups.sh` would be available until Kubernetes 1.30, However, I have verified that the new script is already present in [Kubernetes version 1.28.6 ](https://github.com/googleforgames/agones/pull/3677/files#diff-686b06cb4b89c224b0df5f76fa2bbc7b5757477074e6917a21b8040a9e6857a4R53)

root@deeaacff63a6:/go/src/k8s.io/code-generator# ls
CONTRIBUTING.md  README.md          code-of-conduct.md  generate-groups.sh           go.sum           third_party
LICENSE          SECURITY_CONTACTS  doc.go              generate-internal-groups.sh  kube_codegen.sh  tools.go
OWNERS           cmd                examples            go.mod                       pkg

should I switch to the new script once the PR for the Kubernetes version is merged, or should I make the change within that PR?

**Comment by markmandel** [comment]
> should I switch to the new script once the PR for the Kubernetes version is merged, or should I make the change within that PR?

Let's do it in a new PR once that is merged. One problem at a time 😄
